packages/node_modules/@node-red/editor-client/src/js/ui/editor.js

```diff
@@ -948,6 +948,120 @@ RED.editor = (function() {
         searchInput.trigger("focus");
     }
 
+    function createColorPicker(colorRow, color) {
+        var colorSpec = $("<span/>", {
+            class: "red-ui-subflow-color-spec"
+        }).appendTo(colorRow);
+        var colorButton = $("<label/>").appendTo(colorSpec);
+        var colorIcon = $("<span><i class='fa fa-caret-down'/> </span>").css({
+            width: "30px",
+            height: "20px",
+            "padding-left": "15px",
+            "padding-right": "5px"
+        }).appendTo(colorButton);
+        var colorDisp = $("<label/>").css({
+            width: "40px",
+            height: "20px",
+            margin: "6px",
+            "vertical-align": "middle",
+            "background-color": color
+        }).appendTo(colorButton);
+        var selector = $("<input/>", {
+            id: "red-ui-editor-node-color",
+            type: "text",
+            value: color
+        }).css({
+            width: "150px",
+            border: "none"
+        }).appendTo(colorSpec);
+
+        selector.on("change", function (e) {
+            var color = selector.val();
+            colorDisp.css({
+                "background-color": color
+            });
+        });
+        selector.hover(function (e) {
+            selector.css({
+                "border": "solid 1px #ccc"
+            });
+        }, function() {
+            selector.css({
+                "border": "none"
+            });
+        });
+
+        colorButton.on("click", function (e) {
```

Comment:
FYI - #2236 includes an new api (`RED.popover.panel()`) that can be used as a standard way to create this sort of pop-up dialog. Once that PR is merged, we can migrate the various bits of the UI that do their own thing to use this standard API.

Reply:
Thank you.  I will change this code to use new API after merge of #2236.